lib/record_store/zone.rb

```diff
@@ -189,7 +189,9 @@ def build_authority(authority)
     end
 
     def build_records(records)
-      records.map { |record| Record.build_from_yaml_definition(record) }
+      all_records = records.map { |record| Record.build_from_yaml_definition(record) }
+      config.templates.each { |template| template.inject_records!(all_records: all_records) }
```

Comment:
Any reason to use a named parameter for that method ?

Reply:
I just prefer named parameters because they make multi-argument methods easier to read/use, so i use them out the gate to make refactoring interfaces easier down the road if I ever need to.